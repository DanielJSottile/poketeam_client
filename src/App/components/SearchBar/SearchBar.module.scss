@import '../../../index.scss';

.searchbar,
.search {
  display: flex;
  flex-direction: column;
  margin: auto;
  width: min-content;
  height: 7.7rem;
}

.search {
  justify-content: space-around;
  margin-top: 5%;
}

.label-font {
  font-family: $h2-font;
  color: $super-light-cream;
  font-size: 1rem;
}

.select {
  display: flex;
  flex-direction: row;
  label,
  select {
    font-size: 0.8rem;
  }
  label {
    font-family: $h2-font;
    color: $super-light-cream;
  }
}

.s-input {
  font-family: $body-font;
  font-size: 1rem;
  border: none;
  border-radius: 0.25rem;
}

@media (min-width: 700px) {
  /* Medium Screen */

  .button {
    font-size: 1.3rem;
  }
  .searchbar,
  .search {
    justify-content: center;
    align-items: center;
    flex-direction: row;
    margin: 0;
    width: unset;
    height: unset;
  }

  .button_things,
  .user_things {
    display: inline;
  }

  .mobile-button {
    display: none;
  }
  .label-font {
    font-size: 1.3rem;
  }
  .select {
    display: flex;
    flex-direction: row;
    label,
    select {
      font-size: 1.3rem;
    }
  }
  .s-input {
    font-size: 1.3rem;
  }
}
