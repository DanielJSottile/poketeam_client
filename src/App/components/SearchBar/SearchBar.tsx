import React, { useContext, FunctionComponent, MouseEvent } from 'react';
import {
  faSearch,
  faSort,
  faExclamation,
} from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { TextInput } from '../../@types';
import GeneralContext from '../../contexts/GeneralContext';
import { validateSearch } from '../../utils/validations';
import Button from '../Button';
import Input from '../Input';
import Select from '../Select';
import styles from './SearchBar.module.scss';

type SearchBarProps = {
  /** passed in from Navigation, determines whether to use search or filter */
  isPublic: boolean;
};

const SearchBar: FunctionComponent<SearchBarProps> = ({ isPublic }) => {
  const {
    handleFilter,
    filter,
    filtersort,
    setFilter,
    setFilterSort,
    handleSearch,
    search,
    sort,
    setSearch,
    setSort,
  } = useContext(GeneralContext);

  const searchInput = isPublic ? search : filter;

  const handleSearchOrFilter = (e: MouseEvent<HTMLButtonElement>) => {
    e.preventDefault();
    isPublic ? handleSearch() : handleFilter();
  };

  const setSearchOrFilter = (input: TextInput) =>
    isPublic ? setSearch(input) : setFilter(input);

  const sortOrFilterSort = isPublic ? sort : filtersort;

  const setSortOrFilterSort = (input: TextInput) =>
    isPublic ? setSort(input) : setFilterSort(input);

  return (
    <div className={styles['searchbar']}>
      <form className={styles['search-form']}>
        <div className={styles['search']}>
          <Input
            inputHasError
            isError={!!searchInput.value}
            validationCallback={() => validateSearch(searchInput)}
            containerClass={styles['search-container']}
            inputClass={styles['s-input']}
            labelClass={styles['label-font']}
            label={isPublic ? 'Search: ' : 'Filter: '}
            labelIcon={<FontAwesomeIcon icon={faSearch} />}
            placeholder='e.g. Pikachu'
            value={searchInput.value}
            onChangeCallback={(e) =>
              setSearchOrFilter({ value: e.target.value, touched: true })
            }
            type='text'
            name='search'
            id='search'
          />
          <Select
            selectHasError={false}
            containerClass={styles['select']}
            htmlFor='sort'
            label='Sort By: '
            labelIcon={<FontAwesomeIcon icon={faSort} />}
            name='sort'
            id='sort'
            value={sortOrFilterSort.value}
            onChangeCallback={(e) =>
              setSortOrFilterSort({ value: e.target.value, touched: true })
            }
            options={[
              { value: 'newest', label: 'Newest' },
              { value: 'oldest', label: 'Oldest' },
              { value: 'alphabetical', label: 'A - Z' },
              { value: 'rev alphabetical', label: 'Z - A' },
            ]}
          />
          <Button
            buttonClass={styles['go-button']}
            id='go-button'
            type='submit'
            onClickCallback={(e) => {
              handleSearchOrFilter(e);
            }}
          >
            Go <FontAwesomeIcon icon={faExclamation} />
          </Button>
        </div>
      </form>
    </div>
  );
};

export default SearchBar;
