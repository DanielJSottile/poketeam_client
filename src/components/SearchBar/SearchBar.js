import React, { Component } from 'react'

export default class SearchBar extends Component {

  handleOnSubmit = (e) => {
    e.preventDefault();
    console.log("hey you clicked me");
    const {search, sort} = e.target;

    // some kind of APIservice.getTeams(search.value, sort.hljs-value)
  }

  render() {
    return (
      <div className="searchbar">
        <form 
          className="search"
          onSubmit={this.handleOnSubmit}
        >
          <div>
            <label for="search">Search:</label>
            <input placeholder="e.g. Pikachu" value="" type="text" name="search" id="search" />
            <label for="sort">Sort By:</label>
            <select name="sort" id="sort">
              <option value="Newest">Newest</option>
              <option value="Oldest">Oldest</option>
              <option value="A - Z">A - Z</option>
              <option value="Z - A">Z - A</option>
            </select>
            <button type="submit">Go!</button>
          </div>
        </form>
      </div>
    )
  }
}


