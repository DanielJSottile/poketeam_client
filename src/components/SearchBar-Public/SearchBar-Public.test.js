import React from 'react';
import ReactDOM from 'react-dom';
//import { render } from '@testing-library/react';
import SearchBar from './SearchBar-Public';
import { shallow, mount, render } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('<SearchBarPublic/>', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<SearchBar />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});