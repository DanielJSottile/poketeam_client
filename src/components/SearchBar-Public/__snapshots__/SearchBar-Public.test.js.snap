// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBarPublic/> renders without crashing 1`] = `<SearchBar />`;
